File: array/leetcode/209.go
package leetcode

func minSubArrayLen(target int, nums []int) int {
	i := 0
	result := len(nums) + 1
	sum := 0
	for j := 0; j < len(nums); j++ {
			sum += nums[j]
			if sum >= target {
					sumLength := j - i + 1
					if sumLength < result {
							result = sumLength
					}
					sum -= sum[i]
					i++
			}
	}
	if result == len(nums) + 1 {
			return 0
	} else {
			return 1
	}
}

File: array/cpp/209.cpp
#include <cstdint>
#include <vector>
using namespace std;

class Solution
{
public:
  int minSubArrayLen(int target, vector<int> &nums)
  {
    int i = 0, j = 0;
    int subLength = 0;
    int result = INT32_MAX;
    int sum = 0;
    for (; j < nums.size(); j++)
    {
      sum += nums[j];
      if (sum >= target)
      {
        subLength = j - i + 1;
        result = result < subLength ? result : subLength;
        sum -= nums[i++];
      }
    }
    return result != INT32_MAX ? result : 0;
  }
};